lib/nanoc-conref-fs/conref-fs.rb

```diff
@@ -102,12 +98,32 @@ def parse_with_frontmatter(content_filename)
     ParseResult.new(content: content, attributes: meta, attributes_data: pieces[2])
   end
 
+  # Some of the static class methods below (and elsewhere in the Gem) require 
+  # access to the `data_dir` value from the config. This need comes about
+  # *before* Nanoc has a chance to properly load up the configuration file,
+  # so we're doing a bare-bones load here to gain access to that configuration
+  # item.
+  #
+  # Parameters:
+  # - An optional already-loaded config, so that this can be used later in the
+  #   Nanoc pipeline.
+  #
+  # Returns:
+  # - Custom `data_dir` attribute from the 'conref-fs' data_source
+  # - Default `data_dir` of 'data' if none is configured in `nanoc.yaml`
+  def self.data_dir_name(config=nil)
+    config ||= YAML.load_file('nanoc.yaml')
+    data_sources = config.fetch("data_sources") { [] }
+    data_source = data_sources.find { |ds| ds["type"] == "conref-fs" } || { "data_dir": "data" }
```

Comment:
I am not really a huge fan of this syntax, I find it hard to understand.

Reply:
:100:  Converted to the way I like it too.  :)